packages/dashboard/src/components/csvDownloadButton/useViewportData.ts

```diff
@@ -46,34 +49,41 @@ export const useViewportData = ({
   const viewport = passedInViewport || injectedViewport || DEFAULT_VIEWPORT;
 
   // flatten all the data in a single dataStream into one array of CSVDownloadObject
-  const flattenDataPoints = (dataStream: DataStream, timeOfRequestMS: number) => {
+  const flattenDataPoints = async (dataStream: DataStream, timeOfRequestMS: number) => {
     const { id, unit, resolution, aggregationType, data } = dataStream;
+    const isUnmodeledData = queryConfig.query?.properties?.some((pr) => pr.propertyAlias === id);
+    const { data: unmodeledDescribedTimeSeries } = isUnmodeledData
+      ? await getDescribedTimeSeries({
+          client,
+          alias: id,
+        })
+      : { data: undefined };
 
     return data.reduce((flattenedData: CSVDownloadObject[], currentDataPoint: DataPoint) => {
       const { x: xValue, y: yValue } = currentDataPoint;
       const pointWithinViewport = isTimeWithinViewport(xValue, viewport, timeOfRequestMS);
 
       // do not include data point if it falls outside viewport range
       if (pointWithinViewport) {
-        const isUnmodeledData = queryConfig.query?.properties?.some((pr) => pr.propertyAlias === dataStream.id);
         const assetPropId = !isUnmodeledData ? dataStream.id.split('---') : []; // modeled datastream IDs follow the pattern {assetID}---{propertyID}
         const describedModelProperty = describedAssetsMap[assetPropId[0]]?.properties.find(
-          (p) => p.propertyId === assetPropId[1]
+          ({ propertyId }) => propertyId === assetPropId[1]
         );
 
         const flatDataPoint: CSVDownloadObject = {
```

Comment:
super nit: instead of having inline ternary ops. better to have something lie ` if(unmodeled) { flatDataPoint = {..common, ...} else { flatDataPoint = {...common, ... }}}`

Reply:
great idea, will update code to follow this pattern
will def be much cleaner to read